### ISSUE TITLE
Connect-Graph -Scopes "AuditLog.Read.All" Fails

### ISSUE THREAD

**Issue by ghost** [microsoftgraph/msgraph-sdk-powershell#108]
When using ```Get-MgAuditLogSignIn -Top 10``` I got the following error:

```
Get-MgAuditLogSignIn : Application missing MSGraph API 'Read all audit log data' permission
+ Get-MgAuditLogSignIn -Top 10
+ CategoryInfo          : InvalidOperation: ({ Top = 10, Skip = \u2026ect = , Expand =  }:<>f__AnonymousType2`8) [Get-MgAuditLogSignIn_List], Exception
+ FullyQualifiedErrorId : Authentication_MSGraphPermissionMissing,Microsoft.Graph.PowerShell.Cmdlets.GetMgAuditLogSignIn_List
```

To resolve that error I tried running ```Connect-Graph -Scopes "AuditLog.Read.All"``` but it isn't working and gives the following error:

```
Connect-Graph : Verification code expired before contacting the server
At line:1 char:1
+ Connect-Graph -Scopes "AuditLog.Read.All"
+ CategoryInfo          : NotSpecified: (:) [Connect-Graph], MsalClientException
+ FullyQualifiedErrorId : Microsoft.Identity.Client.MsalClientException,Microsoft.Graph.PowerShell.Authentication.Cmdlets.ConnectGraph
```

I haven't been able to figure out any way to get the ```AuditLog.Read.All``` permission to work.

**Comment by AlanMcBee** [comment]
It's not just that cmdlet. I am having a similar issue with any attempt to re-connect with a different scope than I initially connected with. In my case, I'm trying to add a `GroupMember.Read.All` scope.

**Comment by peombwa** [comment]
Thanks for informing us of this issue. I've been able to repo this on my end, and I can indeed verify that this is a bug. This occurs because the underlaying auth client app is set to authenticate both Microsoft personal accounts and work or school accounts, yet the scopes being passed require an admin consent which is not possible with MSA accounts.

I've opened an issue https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1631 with the MSAL team to advice/provide a fix.

**Comment by peombwa** [comment]
@russellds2w, @AlanMcBee 
For [permissions](https://docs.microsoft.com/en-us/graph/permissions-reference#group-permissions) that require tenant admin consent e.g. `GroupMember.Read.All` or `AuditLog.Read.All`, a tenant ID or domain must be passed to `Connect-Graph -Scopes "GroupMember.Read.All", "AuditLog.Read.All" -TenantId "{TENANT_ID}"` in order for the admin to consent.

The current error message that's returned in not that informative, and it will be resolved in the next release https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/67.

**Comment by rbalsleyMSFT** [comment]
@peombwa When attempting to authenticate via a cert and specifying -tenantID in GUID format, I get this error. If I specify the my tenant domain name, it works fine.

```
Connect-Graph : Error converting value "AuditLog.Read.All" to type 'System.String[]'. Path 'roles', line 1, position 449.
At line:1 char:1
+ Connect-Graph -ClientId ffb9369a-2404-47b0-b2c0-c2667cd29905 -TenantI ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Connect-Graph], JsonSerializationException
    + FullyQualifiedErrorId : Newtonsoft.Json.JsonSerializationException,Microsoft.Graph.PowerShell.Authentication.Cmdlets.ConnectGraph
```

**Comment by darrelmiller** [comment]
@rbalsleyMSFT The use of certificates for authentication is only used for "Application" permissions and not "Delegated" permissions.  When using "Application" permissions it is required to consent to these via the Azure portal.  Certificates are used for non-interactive scenarios and so it would not make sense to pop up a consent dialog.

**Comment by rbalsleyMSFT** [comment]
Hi @darrelmiller thanks. That's understood. However when connecting with cert auth you still need to specify the tenant. At the time, when I was specifying the tenant in GUID format, I was getting the error I mentioned above and wasn't getting it if I passed in my actual tenant domain name. However tonight when using GUID, it's fine.

**Comment by peombwa** [comment]
@rbalsleyMSFT The issue you are referring to is related to https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/154. Check my response there.

**Comment by o-l-a-v** [comment]
Is this fixed yet?

Unbelievable that it's thid hard to get AzureAD signin logs without writing your own PS module against MS Graph. Which I have. But anyways.

* AzureAD does not have any cmdlets for this
* AzureADPreview ```Get-AzureADAuditSignInLogs``` cmdlet is flawed, only returns 1040 objects max, no paging except filtering on date with 15 minute intervals for huge customers
* Microsoft.Graph ```Get-MgAuditLogSignIn``` gives error no matter what #290
* MsOnline has no such cmdlet. But is still the best module for license and MFA information. Even though it's deprecated.

**Comment by peombwa** [comment]
This has already been resolved. `Update-Module Microsoft.Graph` to get the latest.

**Comment by MoAntri** [comment]
Hello,

I have same issue with the command "Get-MgAuditLogSignIn" for the version 0.9.1 of the Module Microsoft.graph.Identity.Auditlogs

**Comment by tioneb22** [comment]
> @russellds2w, @AlanMcBee For [permissions](https://docs.microsoft.com/en-us/graph/permissions-reference#group-permissions) that require tenant admin consent e.g. `GroupMember.Read.All` or `AuditLog.Read.All`, a tenant ID or domain must be passed to `Connect-Graph -Scopes "GroupMember.Read.All", "AuditLog.Read.All" -TenantId "{TENANT_ID}"` in order for the admin to consent.
> 
> The current error message that's returned in not that informative, and it will be resolved in the next release #67.

I stumbled upon this issue when trying to query Get-MgReportAuthenticationMethodUserRegistrationDetail with Powershell. I like the above states, you need to specify all scopes that your Graph API call needs access to, in my case it was Audit.Read.All. Therefore, when connecting to Graph I had to specify like so: _**Connect-Graph -Scopes "AuditLog.Read.All"**_

Query works now, thanks!

**Comment by JeevaRP-SDE** [comment]
Hi @peombwa  @darrelmiller  @MoAntri 
I have a question for , is that possible Can I give permission my app n through C# Code I am trying but not able to Achive that.
 API permission , example auditlog.read , is enable and give grant admin consent through my API C# Code. If it possible means can you give me guide documents

![testperm](https://github.com/microsoftgraph/msgraph-sdk-powershell/assets/68347006/caef1a8e-40cb-404c-83c3-72114c61171a)
